### ISSUE TITLE
Fail to build on m1 macOS.

### ISSUE THREAD

**Issue by capric98** [EleonoreMizo/fmtconv#30]
MacBook Air M1, macOS Monterey, fmtconv r26
```bash
cd build/unix/
./autogen.sh
./configure
make

...=>

  CXX      ../../src/fmtcl/Dither.lo
 In file included from ../../src/fmtcl/Dither.cpp:29:
 In file included from ./../../src/fmtcl/Dither.h:26:
 In file included from ./../../src/conc/ObjPool.h:44:
 In file included from ./../../src/conc/CellPool.h:33:
 In file included from ./../../src/conc/LockFreeStack.h:45:
 In file included from ./../../src/conc/AtomicPtrIntPair.h:40:
 In file included from ./../../src/conc/Interlocked.h:132:
 ./../../src/conc/Interlocked.hpp:267:4: error: invalid output constraint '+A' in asm
         :       "+A" (old)
                 ^
 1 error generated.
 make: *** [../../src/fmtcl/Dither.lo] Error 1
```
It is located at:
https://github.com/EleonoreMizo/fmtconv/blob/ece0a6cd4e69bfc5f7128af9fe3530722f9be77e/src/conc/Interlocked.hpp#L261-L272

**Comment by EleonoreMizo** [comment]
Unfortunately I haven’t access to an ARM Mac to test. Could you please try the following change? In `fmtconv/src/conc/Interlocked.hpp`, replace the `#if` at line 261 with:

```CPP
#if (defined (__APPLE__) && conc_ARCHI == conc_ARCHI_X86) || (defined (__CYGWIN__) && conc_WORD_SIZE == 64)
```
then replace the `__sync_val_compare_and_swap()` at line 312 with:

```CPP
	old = comp;
	__atomic_compare_exchange_n (
		&dest, &old, excg,
		false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST
	);
```

and tell me if it compiles, then works.

**Comment by capric98** [comment]
Now it can compile successfully.
Did a simply test using:
```python3
# src -> YUV420P8
src16 = core.fmtc.bitdepth(clip=src, bits=16)
output = core.fmtc.bitdepth(clip=src16, bits=10, dmode=3)
```
And the output is YUV420P10, seems it works.

Thanks for your fix!

**Comment by EleonoreMizo** [comment]
Thanks for the report. Now fixed in 9fcb86948039f1774b21b11dce94674cef409fc4.
